packages/core/src/types.ts

```diff
@@ -936,14 +951,18 @@ export type Assigner<TContext, TEvent extends EventObject> = (
   meta: AssignMeta<TContext, TEvent>
 ) => Partial<TContext>;
 
+export type PartialAssigner<
```

Comment:
I kinda feel like it would be more appropriate to name this a `PropertyAssigner` and the other one a `PartialAssigner` 😅

Reply:
I know... we can change this for v5